dot/network/message.go

```diff
@@ -36,7 +36,7 @@ type NotificationsMessage interface {
 	Message
 	Type() byte
 	Hash() (common.Hash, error)
-	IsHandshake() bool
+	IsValidHandshake() bool
```

Comment:
Do we still need an interface for this? Could this be refactored further using generics? As it stands, all handshake messages implement every method of this interface except `Hash()`.